pkg/tfgen/source.go

```diff
@@ -98,22 +93,41 @@ func (gh *gitRepoSource) getFile(
 		possibleMarkdownNames = append(possibleMarkdownNames, info.Source)
 	}
 
-	markdownBytes, markdownFileName, found := readMarkdown(repoPath, kind, possibleMarkdownNames)
-	if !found {
-		return nil, nil
-	}
+	return readMarkdown(repoPath, kind, possibleMarkdownNames)
+}
 
-	return &DocFile{Content: markdownBytes, FileName: markdownFileName}, nil
+// An error that represents a missing repo path directory.
+type GetRepoPathErr struct {
+	Expected   string
+	Underlying error
+}
+
+func (e GetRepoPathErr) Error() string {
+	return fmt.Sprintf("Unable to access repository at %s: %s", e.Expected, e.Underlying.Error())
+}
+
+func (e GetRepoPathErr) Unwrap() error {
+	return e.Underlying
 }
 
 var repoPaths sync.Map
 
-func getRepoPath(gitHost string, org string, provider string, version string) (string, error) {
+func getRepoPath(gitHost string, org string, provider string, version string) (_ string, err error) {
 	moduleCoordinates := fmt.Sprintf("%s/%s/terraform-provider-%s", gitHost, org, provider)
 	if version != "" {
 		moduleCoordinates = fmt.Sprintf("%s/%s", moduleCoordinates, version)
 	}
 
+	defer func() {
+		if err == nil {
+			return
+		}
+		err = GetRepoPathErr{
+			Expected:   moduleCoordinates,
+			Underlying: err,
+		}
+	}()
```

Comment:
I find this a bit opaque (vs just having a an inner function that returns vanilla errors and then wrapping them), but I assume this is just some sort of go convention, and if that's the case I can live with it.